### PR TITLE
OADP-987: Add VSM restore plugins

### PR DESCRIPTION
**Pull request by eemcmullan**

### CODE REVIEW

File: internal/restore/volumesnapshotclass_action.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+func (p *VolumeSnapshotClassRestoreItemAction) Execute(input *velero.RestoreItemActionExecuteInput) (*velero.RestoreItemActionExecuteOutput, error) {
+	p.Log.Info("Starting VolumeSnapshotClassRestoreItemAction")
+
+	var snapClass snapshotv1api.VolumeSnapshotClass
+
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(input.Item.UnstructuredContent(), &snapClass); err != nil {
+		return &velero.RestoreItemActionExecuteOutput{}, errors.Wrapf(err, "failed to convert input.Item from unstructured")
+	}
+
+	// check whether or not this VSClass is for blocking VSR until completed
```
Comment (shubham-pampattiwar) [inline internal/restore/volumesnapshotclass_action.go:52]: +1

**Comment by weshayutin** [comment]
@sseago @kaovilai please review

**Comment by shubham-pampattiwar** [comment]
I believe this PR still needs changes for deleting the Volsync VSC once the Velero VSC is ready with the Volsync snaphandle.

**Comment by eemcmullan** [comment]
@weshayutin @shubham-pampattiwar  We can leave the hold on this for now until https://github.com/migtools/volume-snapshot-mover/pull/189 merges, as then I will have to update the VSM dep here to include a final step.

**Comment by eemcmullan** [comment]
@shubham-pampattiwar hold can be lifted. PR includes deleting the extra snapcontent now.

**Comment by kaovilai** [comment]
looks like need to bump to go1.19 for unit test here

**Comment by kaovilai** [comment]
https://github.com/openshift/release/pull/36618

**Review by kaovilai — APPROVED** [review]

**Comment by eemcmullan** [comment]
@shubham-pampattiwar @weshayutin can we please remove the hold on this PR?

**Review by shubham-pampattiwar — APPROVED** [review]

**Pull request merged**